### ISSUE TITLE
Skeleton2D broken

### ISSUE THREAD

**Issue by mortalisk** [godotengine/godot#89986]
### Tested versions

godot 4.2.1, macOS 14.3.1 on macbook pro M2 max

### System information

 macOS 14.3.1 on macbook pro M2 max, forward and mobile renderer tested

### Issue description

Trying to go though the tutorial in the docs, I had issues. I already made an issue for the documentation (it still needs a few correction, allthough it is not as bad as I thought) here: https://github.com/godotengine/godot-docs/issues/9150

The bug is related to rotation of the bones. Both the bones themselves and the polygons/images are not rotated correctly. In godot 3 this works as expected.

<img width="411" alt="Screenshot 2024-03-28 at 15 31 14" src="https://github.com/godotengine/godot/assets/596662/e9d98211-db6c-4877-bfe4-a4031ab68521">

### Steps to reproduce

- Simply follow the turotrial with v3.2.1
- Observe issues with rotation.

### Minimal reproduction project (MRP)

N/A

**Comment by JekSun97** [comment]
Try running [this demo](https://github.com/godotengine/godot-demo-projects/tree/4.2/2d/skeleton) and see the result, maybe you did something wrong during the learning process

**Comment by mortalisk** [comment]
Yeah, things are actually working. I tested some more. What I seem to have done, is to use the rotation and scaling on the bones to position them correctly. It looks correct, until I start using the skeleton, and then it does not work. So, I guess it is not a but, but it certainly is confusing.

**Comment by nomad3k** [comment]
I've seen this issue caused by setting the Skew to any given value and then reverting the Skew to 0.  Some parts remain skewed.
